apps/babelstream/package.py

```diff
@@ -0,0 +1,268 @@
+        if "+acc" in spec:
+            args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+            if("cuda_arch" in spec.variants):
+                cuda_arch_list = spec.variants['cuda_arch'].value
+                # the architecture value is only number so append sm_ to the name
+                cuda_arch = "cc" + cuda_arch_list[0]
+                args.append("-DTARGET_DEVICE=gpu")
+                args.append("-DCUDA_ARCH="+ spec.variants['cuda_arch'].value)
+            elif("cpu_arch" in spec.variants):
+                cpu_arch_list = spec.variants['cpu_arch'].value
+                # the architecture value is only number so append sm_ to the name
+                cpu_arch = cpu_arch_list[0]
+                args.append("-DTARGET_DEVICE=multicore")
+                args.append("-DTARGET_PROCESSOR=" + spec.variants['cpu_arch'].value)
+            
+            #TODO:  include an option to choose the target and the related architecture
+            #       right now we have 2 automatic ones but it could be different too
+            
+
+        #===================================
+        #    STDdata,STDindices,STDranges 
+        #===================================
+        if "+stddata" or "+stdindices" or "+stdranges" in spec:
+            args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+
+        #===================================
+        #             CUDA 
+        #===================================
+        
+        if '+cuda' in spec:
+            # Set up the cuda macros needed by the build
+            cuda_arch_list = spec.variants['cuda_arch'].value
+            # the architecture value is only number so append sm_ to the name
+            cuda_arch = "sm_" + cuda_arch_list[0]
+            args.append("-DCUDA_ARCH=" + cuda_arch)
+            cuda_dir = self.spec["cuda"].prefix
+            cuda_comp = cuda_dir + "/bin/nvcc"
+            args.append("-DCMAKE_CUDA_COMPILER=" + cuda_comp)
+            args.append("-DMEM=" + spec.variants['mem'].value)
+            if(self.spec.variants["flags"].value !='none'):
+                args.append("-DCUDA_EXTRA_FLAGS=" + self.spec.variants["flags"].value )
+
+            
+
+        #===================================
+        #             OMP 
+        #===================================
+        if '+omp' in spec:
+            if("cuda_arch" in spec.variants):
+                cuda_arch_list = spec.variants['cuda_arch'].value
+                # the architecture value is only number so append sm_ to the name
+                cuda_arch = "sm_" + cuda_arch_list[0]
+                args.append("-DOFFLOAD=ON" + " NVIDIA:" + cuda_arch)
+            elif("amdgpu_target" in spec.variants):
+                rocm_arch = spec.variants['amdgpu_target'].value
+                # the architecture value is only number so append sm_ to the name
+                args.append("-DOFFLOAD=ON" +" AMD:"+ rocm_arch)
+            # else:
```

Comment:
We should also support Intel OpenMP offload. You can test this on the NUC in the Zoo.

Reply:
So the commented line actually at #L166 solves this I will uncomment it